### PR TITLE
PYTHON-4662 - Capture Async PyMongo metadata

### PR DESCRIPTION
**Pull request by NoahStapp**

### CODE REVIEW

File: pymongo/client_options.py [hunk 1]
```diff
@@ -144,7 +144,11 @@ def _parse_ssl_options(options: Mapping[str, Any]) -> tuple[Optional[SSLContext]
 
 
 def _parse_pool_options(
-    username: str, password: str, database: Optional[str], options: Mapping[str, Any]
+    username: str,
+    password: str,
+    database: Optional[str],
+    options: Mapping[str, Any],
+    async_client: bool,
```
Comment (blink1073) [inline pymongo/client_options.py:151]: Can we call is `is_async`?

File: test/asynchronous/test_client.py [hunk 2]
```diff
@@ -1938,6 +1940,7 @@ def test_sigstop_sigcont(self):
     async def _test_handshake(self, env_vars, expected_env):
         with patch.dict("os.environ", env_vars):
             metadata = copy.deepcopy(_METADATA)
+            metadata["driver"]["name"] = "AsyncPyMongo"
```
Comment (ShaneHarvey) [inline test/asynchronous/test_client.py:1943]: Could we use "PyMongo-async" and add a test for DriverInfo with AsyncMongoClient?

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: pymongo/asynchronous/mongo_client.py [hunk 3]
```diff
@@ -830,7 +830,10 @@ def __init__(
         # Username and password passed as kwargs override user info in URI.
         username = opts.get("username", username)
         password = opts.get("password", password)
-        self._options = options = ClientOptions(username, password, dbase, opts)
+        if _IS_SYNC:
+            self._options = options = ClientOptions(username, password, dbase, opts, False)
```
[the remark this answers is not in the document]
Comment (NoahStapp) [reply pymongo/asynchronous/mongo_client.py:834]: That's a much better idea, thanks!

**Comment by ShaneHarvey** [comment]
Here's what Java looks like:
```
{"name":"mongo-java-driver|reactive-streams","version":"4.7.1"}
```

Given that, I propose we use `PyMongo|async`.

**Comment by ShaneHarvey** [comment]
Yes `PyMongo|async|driverName` is correct.

File: pymongo/client_options.py [hunk 4]
```diff
@@ -144,7 +144,11 @@ def _parse_ssl_options(options: Mapping[str, Any]) -> tuple[Optional[SSLContext]
 
 
 def _parse_pool_options(
-    username: str, password: str, database: Optional[str], options: Mapping[str, Any]
+    username: str,
+    password: str,
+    database: Optional[str],
+    options: Mapping[str, Any],
+    is_async: bool,
```
Comment (ShaneHarvey) [inline pymongo/client_options.py:151]: Given we use `_IS_SYNC` everywhere, can we rename this to "is_sync" for consistency?

Comment (NoahStapp) [reply pymongo/client_options.py]: Done!

**Review by blink1073 — APPROVED** [review]
LGTM!

**Review by ShaneHarvey — APPROVED** [review]
LGTM

**Pull request merged**